diff -Naur httpd-2.0.52/docs/cgi-examples/printenv httpd-2.0.52-perlfix/docs/cgi-examples/printenv
--- httpd-2.0.52/docs/cgi-examples/printenv	2000-03-31 08:32:10.000000000 +0200
+++ httpd-2.0.52-perlfix/docs/cgi-examples/printenv	2004-10-26 20:20:45.509875832 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 ##
 ##  printenv -- demo CGI program which just prints its environment
 ##
diff -Naur httpd-2.0.52/srclib/apr/build/lineends.pl httpd-2.0.52-perlfix/srclib/apr/build/lineends.pl
--- httpd-2.0.52/srclib/apr/build/lineends.pl	2002-02-21 06:36:44.000000000 +0100
+++ httpd-2.0.52-perlfix/srclib/apr/build/lineends.pl	2004-10-26 20:20:51.687936624 +0200
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 #
 #  Heuristically converts line endings to the current OS's preferred format
 #  
diff -Naur httpd-2.0.52/srclib/apr/helpers/apr_rename.pl httpd-2.0.52-perlfix/srclib/apr/helpers/apr_rename.pl
--- httpd-2.0.52/srclib/apr/helpers/apr_rename.pl	2002-11-30 17:34:41.000000000 +0100
+++ httpd-2.0.52-perlfix/srclib/apr/helpers/apr_rename.pl	2004-10-26 20:20:52.138868072 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/perl
 use strict;
 use ExtUtils::MakeMaker qw(prompt);
 use File::Find;
diff -Naur httpd-2.0.52/srclib/apr-util/build/w32locatedb.pl httpd-2.0.52-perlfix/srclib/apr-util/build/w32locatedb.pl
--- httpd-2.0.52/srclib/apr-util/build/w32locatedb.pl	2003-11-10 23:08:14.000000000 +0100
+++ httpd-2.0.52-perlfix/srclib/apr-util/build/w32locatedb.pl	2004-10-26 20:20:53.244699960 +0200
@@ -1,4 +1,4 @@
-#! perl -w
+#!/usr/bin/perl
 #
 # w32locatedb.pl -- Build apr-util with Berkeley DB on Win32
 #
diff -Naur httpd-2.0.52/srclib/pcre/perltest httpd-2.0.52-perlfix/srclib/pcre/perltest
--- httpd-2.0.52/srclib/pcre/perltest	2002-03-20 06:54:05.000000000 +0100
+++ httpd-2.0.52-perlfix/srclib/pcre/perltest	2004-10-26 20:20:52.904751640 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#!/usr/bin/perl
 
 # Program for testing regular expressions with perl to check that PCRE handles
 # them the same.
diff -Naur httpd-2.0.52/srclib/pcre/perltest8 httpd-2.0.52-perlfix/srclib/pcre/perltest8
--- httpd-2.0.52/srclib/pcre/perltest8	2002-03-20 06:54:05.000000000 +0100
+++ httpd-2.0.52-perlfix/srclib/pcre/perltest8	2004-10-26 20:20:52.821764256 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#!/usr/bin/perl
 
 # Program for testing regular expressions with perl to check that PCRE handles
 # them the same. This is the version that supports /8 for UTF-8 testing. It
diff -Naur httpd-2.0.52/support/SHA1/convert-sha1.pl httpd-2.0.52-perlfix/support/SHA1/convert-sha1.pl
--- httpd-2.0.52/support/SHA1/convert-sha1.pl	1999-08-24 08:45:59.000000000 +0200
+++ httpd-2.0.52-perlfix/support/SHA1/convert-sha1.pl	2004-10-26 20:20:53.521657856 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/perl
 use strict;
 
 # This is public domain code.  Do whatever you want with it.
diff -Naur httpd-2.0.52/support/SHA1/htpasswd-sha1.pl httpd-2.0.52-perlfix/support/SHA1/htpasswd-sha1.pl
--- httpd-2.0.52/support/SHA1/htpasswd-sha1.pl	1999-08-24 08:45:59.000000000 +0200
+++ httpd-2.0.52-perlfix/support/SHA1/htpasswd-sha1.pl	2004-10-26 20:20:53.538655272 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/perl
 use strict;
 #
 # Utility which takes a username and password
diff -Naur httpd-2.0.52/support/apxs.in httpd-2.0.52-perlfix/support/apxs.in
--- httpd-2.0.52/support/apxs.in	2004-02-09 21:59:49.000000000 +0100
+++ httpd-2.0.52-perlfix/support/apxs.in	2004-10-26 20:20:53.753622592 +0200
@@ -1,4 +1,4 @@
-#!@perlbin@ -w
+#!/usr/bin/perl
 #
 # Copyright 1999-2004 The Apache Software Foundation
 #
diff -Naur httpd-2.0.52/support/dbmmanage.in httpd-2.0.52-perlfix/support/dbmmanage.in
--- httpd-2.0.52/support/dbmmanage.in	2004-02-09 21:59:49.000000000 +0100
+++ httpd-2.0.52-perlfix/support/dbmmanage.in	2004-10-26 20:20:53.673634752 +0200
@@ -1,4 +1,4 @@
-#!@perlbin@
+#!/usr/bin/perl
 #
 # Copyright 2001-2004 The Apache Software Foundation
 #
diff -Naur httpd-2.0.52/support/list_hooks.pl httpd-2.0.52-perlfix/support/list_hooks.pl
--- httpd-2.0.52/support/list_hooks.pl	2004-02-09 21:59:49.000000000 +0100
+++ httpd-2.0.52-perlfix/support/list_hooks.pl	2004-10-26 20:20:53.559652080 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/perl
 #
 # Copyright 2001-2004 The Apache Software Foundation
 #
diff -Naur httpd-2.0.52/support/log_server_status.in httpd-2.0.52-perlfix/support/log_server_status.in
--- httpd-2.0.52/support/log_server_status.in	2004-02-09 21:59:49.000000000 +0100
+++ httpd-2.0.52-perlfix/support/log_server_status.in	2004-10-26 20:20:53.706629736 +0200
@@ -1,4 +1,4 @@
-#!@perlbin@
+#!/usr/bin/perl
 #
 # Copyright 2001-2004 The Apache Software Foundation
 #
diff -Naur httpd-2.0.52/support/logresolve.pl.in httpd-2.0.52-perlfix/support/logresolve.pl.in
--- httpd-2.0.52/support/logresolve.pl.in	2004-02-09 21:59:49.000000000 +0100
+++ httpd-2.0.52-perlfix/support/logresolve.pl.in	2004-10-26 20:20:53.655637488 +0200
@@ -1,4 +1,4 @@
-#!@perlbin@
+#!/usr/bin/perl
 #
 # Copyright 2001-2004 The Apache Software Foundation
 #
diff -Naur httpd-2.0.52/support/phf_abuse_log.cgi.in httpd-2.0.52-perlfix/support/phf_abuse_log.cgi.in
--- httpd-2.0.52/support/phf_abuse_log.cgi.in	2004-02-09 21:59:49.000000000 +0100
+++ httpd-2.0.52-perlfix/support/phf_abuse_log.cgi.in	2004-10-26 20:20:53.772619704 +0200
@@ -1,4 +1,4 @@
-#!@perlbin@
+#!/usr/bin/perl
 #
 # Copyright 2001-2004 The Apache Software Foundation
 #
diff -Naur httpd-2.0.52/support/split-logfile.in httpd-2.0.52-perlfix/support/split-logfile.in
--- httpd-2.0.52/support/split-logfile.in	2004-02-09 21:59:49.000000000 +0100
+++ httpd-2.0.52-perlfix/support/split-logfile.in	2004-10-26 20:20:53.736625176 +0200
@@ -1,4 +1,4 @@
-#!@perlbin@
+#!/usr/bin/perl
 #
 # Copyright 2001-2004 The Apache Software Foundation
 #
diff -Naur httpd-2.0.52/test/check_chunked httpd-2.0.52-perlfix/test/check_chunked
--- httpd-2.0.52/test/check_chunked	2004-02-09 21:59:50.000000000 +0100
+++ httpd-2.0.52-perlfix/test/check_chunked	2004-10-26 20:20:53.841609216 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/usr/bin/perl
 #
 # Copyright 2000-2004 The Apache Software Foundation
 #
