---
 support/ab.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/support/ab.c
+++ b/support/ab.c
@@ -2057,7 +2057,7 @@ int main(int argc, const char * const ar
 	    strcpy(content_type, optarg);
 	    break;
 	case 'C':
-            cookie = apr_pstrcat(cntxt, "Cookie: ", optarg, "\r\n", NULL);
+            cookie = apr_pstrcat(cntxt, cookie, "Cookie: ", optarg, "\r\n", NULL);
 	    break;
 	case 'A':
 	    /*
