---
 support/Makefile.in |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/support/Makefile.in
+++ b/support/Makefile.in
@@ -8,6 +8,8 @@ TARGETS  = $(PROGRAMS)
 
 PROGRAM_LDADD        = $(UTIL_LDFLAGS) $(PROGRAM_DEPENDENCIES) $(EXTRA_LIBS) $(AP_LIBS)
 PROGRAM_DEPENDENCIES = 
+LAZYLDFLAGS          = -Wl,-z,now
+
 
 include $(top_builddir)/build/rules.mk
 
@@ -59,7 +61,7 @@ checkgid: $(checkgid_OBJECTS)
 
 suexec_OBJECTS = suexec.lo
 suexec: $(suexec_OBJECTS)
-	$(LINK) $(suexec_OBJECTS)
+	$(LINK) $(LAZYLDFLAGS) $(suexec_OBJECTS)
 
 httxt2dbm_OBJECTS = httxt2dbm.lo
 httxt2dbm: $(httxt2dbm_OBJECTS)
