---
 Makefile.in         |   10 ++++++++--
 support/Makefile.in |   16 +++-------------
 2 files changed, 11 insertions(+), 15 deletions(-)

--- a/Makefile.in
+++ b/Makefile.in
@@ -183,14 +183,20 @@ install-include:
 	@cp -p $(srcdir)/os/$(OS_DIR)/*.h $(DESTDIR)$(includedir)
 	@chmod 644 $(DESTDIR)$(includedir)/*.h
 
+MAN1=
+MAN8=httpd.8 suexec.8 apxs.8
 install-man:
 	@echo Installing man pages and online manual
 	@test -d $(DESTDIR)$(mandir)      || $(MKINSTALLDIRS) $(DESTDIR)$(mandir)
 	@test -d $(DESTDIR)$(mandir)/man1 || $(MKINSTALLDIRS) $(DESTDIR)$(mandir)/man1
 	@test -d $(DESTDIR)$(mandir)/man8 || $(MKINSTALLDIRS) $(DESTDIR)$(mandir)/man8
 	@test -d $(DESTDIR)$(manualdir)   || $(MKINSTALLDIRS) $(DESTDIR)$(manualdir)
-	@cp -p $(top_srcdir)/docs/man/*.1 $(DESTDIR)$(mandir)/man1
-	@cp -p $(top_srcdir)/docs/man/*.8 $(DESTDIR)$(mandir)/man8
+	@for i in $(MAN1) ; do \
+		cp -p $(top_srcdir)/docs/man/$$i $(DESTDIR)$(mandir)/man1; \
+	done
+	@for i in $(MAN8) ; do \
+		cp -p $(top_srcdir)/docs/man/$$i $(DESTDIR)$(mandir)/man8; \
+	done
 	@(cd $(top_srcdir)/docs/manual && cp -rp * $(DESTDIR)$(manualdir))
 	@(cd $(DESTDIR)$(manualdir) && find . -name ".svn" -type d -print | xargs rm -rf 2>/dev/null ) || true
 
--- a/support/Makefile.in
+++ b/support/Makefile.in
@@ -1,9 +1,8 @@
-DISTCLEAN_TARGETS = apxs apachectl dbmmanage log_server_status \
-	logresolve.pl phf_abuse_log.cgi split-logfile envvars-std
+DISTCLEAN_TARGETS = apxs
 
 CLEAN_TARGETS = suexec
 
-PROGRAMS = htpasswd htdigest rotatelogs logresolve ab checkgid htdbm httxt2dbm
+PROGRAMS = apxs
 TARGETS  = $(PROGRAMS)
 
 PROGRAM_LDADD        = $(UTIL_LDFLAGS) $(PROGRAM_DEPENDENCIES) $(EXTRA_LIBS) $(AP_LIBS)
@@ -12,22 +11,13 @@ PROGRAM_DEPENDENCIES = 
 include $(top_builddir)/build/rules.mk
 
 install:
-	@test -d $(DESTDIR)$(bindir) || $(MKINSTALLDIRS) $(DESTDIR)$(bindir)
 	@test -d $(DESTDIR)$(sbindir) || $(MKINSTALLDIRS) $(DESTDIR)$(sbindir)
-	@test -d $(DESTDIR)$(libexecdir) || $(MKINSTALLDIRS) $(DESTDIR)$(libexecdir)
-	@cp -p $(top_builddir)/server/httpd.exp $(DESTDIR)$(libexecdir)
-	@for i in apxs apachectl dbmmanage; do \
+	@for i in $(PROGRAMS) ; do \
 	    if test -f "$(builddir)/$$i"; then \
 	        cp -p $$i $(DESTDIR)$(sbindir); \
 	        chmod 755 $(DESTDIR)$(sbindir)/$$i; \
 	    fi ; \
 	done
-	@if test -f "$(builddir)/envvars-std"; then \
-	    cp -p envvars-std $(DESTDIR)$(sbindir); \
-	    if test ! -f $(DESTDIR)$(sbindir)/envvars; then \
-	        cp -p envvars-std $(DESTDIR)$(sbindir)/envvars ; \
-	    fi ; \
-	fi
 
 htpasswd_OBJECTS = htpasswd.lo
 htpasswd: $(htpasswd_OBJECTS)
