---
 buildconf                 |    2 +-
 srclib/apr-util/buildconf |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/buildconf
+++ b/buildconf
@@ -29,7 +29,7 @@ while test $# -gt 0 
 do
   # Normalize
   case "$1" in
-  -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
+  -*=*) optarg=`export LC_ALL=C; echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
   *) optarg= ;;
   esac
 
--- a/srclib/apr-util/buildconf
+++ b/srclib/apr-util/buildconf
@@ -8,7 +8,7 @@ while test $# -gt 0 
 do
   # Normalize
   case "$1" in
-  -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
+  -*=*) optarg=`export LC_ALL=C; echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
   *) optarg= ;;
   esac
 
