diff -NurpP httpd-2.0.54-peruser/server/mpm/experimental/peruser/peruser.c httpd-2.0.54-peruser-brigadefix/server/mpm/experimental/peruser/peruser.c
--- httpd-2.0.54-peruser/server/mpm/experimental/peruser/peruser.c	2005-08-22 08:04:59.397143536 +0200
+++ httpd-2.0.54-peruser-brigadefix/server/mpm/experimental/peruser/peruser.c	2005-08-22 08:07:32.979795424 +0200
@@ -886,10 +886,8 @@ static int peruser_process_connection(co
             net = filter->ctx;
             net->in_ctx = apr_palloc(conn->pool, sizeof(*net->in_ctx));
             net->in_ctx->b = bb;
-/*
             net->in_ctx->tmpbb = apr_brigade_create(net->in_ctx->b->p, 
                 net->in_ctx->b->bucket_alloc);
-*/
         }
     }
     _DBG("leaving (DECLINED)", 0);
