Customizations for Gentoo
diff -ur httpd-2.2.0/include/httpd.h httpd-2.2.0.patched/include/httpd.h
--- httpd-2.2.0/include/httpd.h	2005-09-29 13:44:53.000000000 -0700
+++ httpd-2.2.0.patched/include/httpd.h	2006-02-04 13:03:05.000000000 -0800
@@ -155,7 +155,7 @@
 
 /** The path to the suExec wrapper, can be overridden in Configuration */
 #ifndef SUEXEC_BIN
-#define SUEXEC_BIN  HTTPD_ROOT "/bin/suexec"
+#define SUEXEC_BIN  "/usr/sbin/suexec2"
 #endif
 
 /** The timeout for waiting for messages */
diff -ur httpd-2.2.0/server/core.c httpd-2.2.0.patched/server/core.c
--- httpd-2.2.0/server/core.c	2005-11-15 11:39:15.000000000 -0800
+++ httpd-2.2.0.patched/server/core.c	2006-02-04 13:07:40.000000000 -0800
@@ -2756,7 +2756,7 @@
         ap_add_version_component(pconf, AP_SERVER_BASEPRODUCT "/" AP_SERVER_MAJORVERSION);
     }
     else {
-        ap_add_version_component(pconf, AP_SERVER_BASEVERSION " (" PLATFORM ")");
+        ap_add_version_component(pconf, AP_SERVER_BASEVERSION " (Gentoo)");
     }
 
     /*
diff -ru httpd-2.2.0/support/apxs.in httpd-2.2.0.patched/support/apxs.in
--- httpd-2.2.0/support/apxs.in	2005-06-10 02:18:14.000000000 -0700
+++ httpd-2.2.0.patched/support/apxs.in	2006-02-14 21:17:47.000000000 -0800
@@ -190,9 +190,6 @@
 my $httpd = get_vars("sbindir") . "/" . get_vars("progname");
 $httpd = eval qq("$httpd");
 $httpd = eval qq("$httpd");
-my $envvars = get_vars("sbindir") . "/envvars";
-$envvars = eval qq("$envvars");
-$envvars = eval qq("$envvars");
 
 #allow apxs to be run from the source tree, before installation
 if ($0 =~ m:support/apxs$:) {
@@ -204,14 +201,6 @@
 	exit 1;
 }
 
-unless (grep /mod_so/, `. $envvars && $httpd -l`) {
-    error("Sorry, no shared object support for Apache");
-    error("available under your platform. Make sure");
-    error("the Apache module mod_so is compiled into");
-    error("your server binary `$httpd'.");
-    exit 1;
-}
-
 sub get_config_vars{
     my ($file, $rh_config) = @_;
 
