Index: configure.in
===================================================================
--- configure.in	(revision 647627)
+++ configure.in	(working copy)
@@ -237,7 +237,7 @@
       LIBTOOL="$my_libtool \$(LTFLAGS)"
       libtoolversion=`$my_libtool --version`
       case $libtoolversion in
-          *1.[[45]]*)
+          *1.[[45]]* | *[[2-9]].[[0-9]]*)
               SH_LIBTOOL='$(LIBTOOL)'
               SHLTCFLAGS="-prefer-pic"
               LTCFLAGS="-prefer-non-pic -static"
