diff -ru httpd-2.0.52/modules/mappers/mod_rewrite.c httpd-2.0.52.ampescape/modules/mappers/mod_rewrite.c
--- httpd-2.0.52/modules/mappers/mod_rewrite.c	2004-08-27 21:23:26.000000000 +0200
+++ httpd-2.0.52.ampescape/modules/mappers/mod_rewrite.c	2004-11-19 19:04:17.190398544 +0100
@@ -964,6 +964,7 @@
         map_pfn_register("tolower", rewrite_mapfunc_tolower);
         map_pfn_register("toupper", rewrite_mapfunc_toupper);
         map_pfn_register("escape", rewrite_mapfunc_escape);
+		map_pfn_register("ampescape", rewrite_mapfunc_ampescape);
         map_pfn_register("unescape", rewrite_mapfunc_unescape);
     }
     return OK;
@@ -3346,6 +3347,29 @@
     return value;
 }
 
+static char *rewrite_mapfunc_ampescape(request_rec *r, char *key) {
+	/* we only need to escape the ampersand */
+
+	unsigned char *copy = (char *)apr_palloc(r->pool, 3 * strlen(key) + 3);
+	const unsigned char *s = (const unsigned char *)key;
+	unsigned char *d = (unsigned char *)copy;
+	unsigned c;
+
+	while ((c = *s)) {
+		if (c == '&') {
+			*d++ = '%';
+			*d++ = '2';
+			*d++ = '6';
+		} else {
+			*d++ = c;
+		}
+		++s;
+	}
+	*d = '\0';
+
+	return copy;
+}
+
 static char *rewrite_mapfunc_unescape(request_rec *r, char *key)
 {
     char *value;
diff -ru httpd-2.0.52/modules/mappers/mod_rewrite.h httpd-2.0.52.ampescape/modules/mappers/mod_rewrite.h
--- httpd-2.0.52/modules/mappers/mod_rewrite.h	2004-08-20 23:17:41.000000000 +0200
+++ httpd-2.0.52.ampescape/modules/mappers/mod_rewrite.h	2004-11-19 18:29:50.000000000 +0100
@@ -389,6 +389,7 @@
 static char *rewrite_mapfunc_toupper(request_rec *r, char *key);
 static char *rewrite_mapfunc_tolower(request_rec *r, char *key);
 static char *rewrite_mapfunc_escape(request_rec *r, char *key);
+static char *rewrite_mapfunc_ampescape(request_rec *r, char *key);
 static char *rewrite_mapfunc_unescape(request_rec *r, char *key);
 
 static char *select_random_value_part(request_rec *r, char *value);
