diff -pruN a/Makefile.in b/Makefile.in
--- a/Makefile.in	2011-08-24 00:34:55.000000000 +0300
+++ b/Makefile.in	2011-08-24 00:29:01.000000000 +0300
@@ -14,7 +14,7 @@ PROGRAM_DEPENDENCIES = \
 
 PROGRAMS        = $(PROGRAM_NAME)
 TARGETS         = $(PROGRAMS) $(shared_build) $(other_targets)
-INSTALL_TARGETS = install-htdocs install-error install-icons \
+INSTALL_TARGETS = install-conf install-htdocs install-error install-icons \
 	install-other install-cgi install-include install-suexec install-build \
 	install-man
 
@@ -202,14 +202,20 @@ install-include:
 	  $(INSTALL_DATA) $$hdr $(DESTDIR)$(includedir); \
 	done
 
+MAN1=
+MAN8=apxs.8 httpd.8 suexec.8
 install-man:
 	@echo Installing man pages and online manual
 	@test -d $(DESTDIR)$(mandir)      || $(MKINSTALLDIRS) $(DESTDIR)$(mandir)
 	@test -d $(DESTDIR)$(mandir)/man1 || $(MKINSTALLDIRS) $(DESTDIR)$(mandir)/man1
 	@test -d $(DESTDIR)$(mandir)/man8 || $(MKINSTALLDIRS) $(DESTDIR)$(mandir)/man8
 	@test -d $(DESTDIR)$(manualdir)   || $(MKINSTALLDIRS) $(DESTDIR)$(manualdir)
-	@cp -p $(top_srcdir)/docs/man/*.1 $(DESTDIR)$(mandir)/man1
-	@cp -p $(top_srcdir)/docs/man/*.8 $(DESTDIR)$(mandir)/man8
+	@for i in $(MAN1) ; do \
+		cp -p $(top_srcdir)/docs/man/$$i $(DESTDIR)$(mandir)/man1; \
+	done
+	@for i in $(MAN8) ; do \
+		cp -p $(top_srcdir)/docs/man/$$i $(DESTDIR)$(mandir)/man8; \
+	done
 	@if test "x$(RSYNC)" != "x" && test -x $(RSYNC) ; then \
 	  $(RSYNC) --exclude .svn -rlpt --numeric-ids $(top_srcdir)/docs/manual/ $(DESTDIR)$(manualdir)/; \
 	else \
diff -pruN a/support/Makefile.in b/support/Makefile.in
--- a/support/Makefile.in	2011-04-16 22:09:47.000000000 +0300
+++ b/support/Makefile.in	2011-08-24 00:33:10.000000000 +0300
@@ -1,9 +1,8 @@
-DISTCLEAN_TARGETS = apxs apachectl dbmmanage log_server_status \
-	logresolve.pl phf_abuse_log.cgi split-logfile envvars-std
+DISTCLEAN_TARGETS = apxs
 
 CLEAN_TARGETS = suexec
 
-PROGRAMS = htpasswd htdigest rotatelogs logresolve ab htdbm htcacheclean httxt2dbm $(NONPORTABLE_SUPPORT)
+PROGRAMS = apxs
 TARGETS  = $(PROGRAMS)
 
 PROGRAM_LDADD        = $(UTIL_LDFLAGS) $(PROGRAM_DEPENDENCIES) $(EXTRA_LIBS) $(AP_LIBS)
@@ -12,22 +11,13 @@ PROGRAM_DEPENDENCIES =
 include $(top_builddir)/build/rules.mk
 
 install:
-	@test -d $(DESTDIR)$(bindir) || $(MKINSTALLDIRS) $(DESTDIR)$(bindir)
 	@test -d $(DESTDIR)$(sbindir) || $(MKINSTALLDIRS) $(DESTDIR)$(sbindir)
-	@test -d $(DESTDIR)$(libexecdir) || $(MKINSTALLDIRS) $(DESTDIR)$(libexecdir)
-	@cp -p $(top_builddir)/server/httpd.exp $(DESTDIR)$(libexecdir)
-	@for i in apxs apachectl dbmmanage; do \
+	@for i in $(PROGRAMS) ; do \
 	    if test -f "$(builddir)/$$i"; then \
 	        cp -p $$i $(DESTDIR)$(sbindir); \
 	        chmod 755 $(DESTDIR)$(sbindir)/$$i; \
 	    fi ; \
 	done
-	@if test -f "$(builddir)/envvars-std"; then \
-	    cp -p envvars-std $(DESTDIR)$(sbindir); \
-	    if test ! -f $(DESTDIR)$(sbindir)/envvars; then \
-	        cp -p envvars-std $(DESTDIR)$(sbindir)/envvars ; \
-	    fi ; \
-	fi
 
 htpasswd_OBJECTS = htpasswd.lo
 htpasswd: $(htpasswd_OBJECTS)
@@ -60,7 +50,7 @@ checkgid: $(checkgid_OBJECTS)
 
 suexec_OBJECTS = suexec.lo
 suexec: $(suexec_OBJECTS)
-	$(LINK) $(suexec_OBJECTS)
+	$(LINK) -Wl,-z,now $(suexec_OBJECTS)
 
 htcacheclean_OBJECTS = htcacheclean.lo
 htcacheclean: $(htcacheclean_OBJECTS)
