diff -ur apache_1.3.34.orig/src/modules/standard/mod_dir.c apache_1.3.34/src/modules/standard/mod_dir.c
--- apache_1.3.34.orig/src/modules/standard/mod_dir.c	2004-11-24 11:10:19.000000000 -0800
+++ apache_1.3.34/src/modules/standard/mod_dir.c	2006-04-01 15:09:24.000000000 -0800
@@ -30,6 +30,7 @@
 
 typedef struct dir_config_struct {
     array_header *index_names;
+	array_header *index_names_incr;
 } dir_config_rec;
 
 #define DIR_CMD_PERMS OR_INDEXES
@@ -45,11 +46,25 @@
     return NULL;
 }
 
+static const char *add_index_incr(cmd_parms *cmd, void *dummy, char *arg)
+{
+    dir_config_rec *d = dummy;
+
+    if (!d->index_names_incr) {
+	d->index_names_incr = ap_make_array(cmd->pool, 2, sizeof(char *));
+    }
+    *(char **)ap_push_array(d->index_names_incr) = arg;
+    return NULL;
+}
+
 static const command_rec dir_cmds[] =
 {
     {"DirectoryIndex", add_index, NULL,
      DIR_CMD_PERMS, ITERATE,
      "a list of file names"},
+    {"AddDirectoryIndex", add_index_incr, NULL,
+     DIR_CMD_PERMS, ITERATE,
+     "a list of file names"},
     {NULL}
 };
 
@@ -59,6 +74,7 @@
     (dir_config_rec *) ap_pcalloc(p, sizeof(dir_config_rec));
 
     new->index_names = NULL;
+	new->index_names_incr = NULL;
     return (void *) new;
 }
 
@@ -68,7 +84,26 @@
     dir_config_rec *base = (dir_config_rec *) basev;
     dir_config_rec *add = (dir_config_rec *) addv;
 
-    new->index_names = add->index_names ? add->index_names : base->index_names;
+
+	if (add->index_names) {
+		new->index_names = add->index_names;
+	} else {
+		new->index_names = base->index_names;
+		if (base->index_names_incr) {
+			if (!new->index_names) {
+				new->index_names = ap_make_array(p, 2, sizeof(char *));
+			}
+			ap_array_cat(new->index_names, base->index_names_incr);
+		}
+	}
+
+	if (add->index_names_incr) {
+		if (!new->index_names) {
+			new->index_names = ap_make_array(p, 2, sizeof(char *));
+		}
+		ap_array_cat(new->index_names, add->index_names_incr);
+	}
+	
     return new;
 }
 
@@ -105,6 +140,14 @@
         r->filename = ap_pstrcat(r->pool, r->filename, "/", NULL);
     }
 
+	if (d->index_names_incr) {
+		if (!d->index_names) {
+			d->index_names = d->index_names_incr;
+		} else {
+			ap_array_cat(d->index_names, d->index_names_incr);
+		}
+	}
+
     if (d->index_names) {
 	names_ptr = (char **)d->index_names->elts;
 	num_names = d->index_names->nelts;
